gateway/engine/es/src/main/java/io/apiman/gateway/engine/es/ESMetrics.java

```diff
@@ -96,13 +110,22 @@ public void run() {
      */
     protected void processQueue() {
         try {
-            RequestMetric metric = queue.take();
-            Index index = new Index.Builder(metric).refresh(false)
-                    .index(getIndexName())
-                    .type("request").build(); //$NON-NLS-1$
-            DocumentResult result = getClient().execute(index);
+            Collection<RequestMetric> batch = new ArrayList<>(this.batchSize);
+            RequestMetric rm = queue.take();
+            batch.add(rm);
+			queue.drainTo(batch, this.batchSize - 1);
+			
+			Builder builder = new Bulk.Builder();
```

Comment:
What happens if you're below the batch threshold  but no new stuff comes? Is there a timer to also flush it out? e.g. after 5 minutes. I made a timer component you could use if that seems useful.